src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -442,11 +443,11 @@ private void insertSyncInfoIfNotPresent(long epochSeconds, ThingShadowSyncConfig
                 .build());
     }
 
-    private Optional<ThingShadowSyncConfiguration> getCoreThingShadowSyncConfiguration(String thingName) {
+    private List<ThingShadowSyncConfiguration> getCoreThingShadowSyncConfiguration(String thingName) {
```

Comment:
will this ever be a list that is not either length 0 or 1? why did you have to change it from optional to list?

Reply:
In the current code, we're using the [same method](https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/main/src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java#L213-L214) while updating the thing shadow sync configuration changes. So, we need a list instead of optional to update all the shadows. Description of the PR has better context with an example.